torch_geometric/data/data.py

```diff
@@ -598,6 +598,11 @@ def subgraph(self, subset: Tensor) -> 'Data':
         else:
             num_nodes = subset.size(0)
 
+            # check if selection is unordered and re-map edge_index if necessary
+            if (subset[:-1] > subset[1:]).any():
+                node_idx = torch.argsort(subset)
+                edge_index = node_idx[edge_index]
```

Comment:
Would simply calling `torch.sort` at the start of the function resolve this issue?

Reply:
This would fix the bug, too. 

The solution here has the additional utility of allowing a user to change the order of vertices to a new order of their choice. If this is not considered useful for these methods, we could also simply switch to always sorting all integer lists.

Reply:
Users can change the order outside of this function if needed. So lets switch to sorting.

Reply:
Fair enough. I have switched to using torch.unique at the beginning of both methods. This also addresses my question regarding handling duplicate indices.

Perhaps we should add a `reorder_nodes` method as a utility function? But this is a separate PR I guess.